### DISCUSSION TITLE
user customizable game replay playback speed

### DISCUSSION THREAD

**Discussion by bcambl** [BattlesnakeOfficial/feedback#134] [Feature Requests]
I noticed recently that the playback speed has been increased. I am unsure if this was a planned change, but thought it would be a nice feature to allow users to configure the playback speed of the replays.

Perhaps a 1x/2x/4x option or a slider. Either on a per-game basis on the UI or at the profile level.

**Comment by originalwebgurl** [comment]
Today's your lucky day @bcambl! We've recently started to work on this very feature. 

An initial version was released today incognito [Board - v1.1.1](https://github.com/BattlesnakeOfficial/board/releases/tag/1.1.1) 

You're welcome to test it out and provide feedback. I'd love to get more input as we develop this. It's not accessible via the UI yet, but you can get there using the browser dev tools by adding the new /settings route to a game page:
![board-settings-initial](https://user-images.githubusercontent.com/2898748/139942699-b6b56746-65e6-43bb-9cc5-5b02fd27929b.gif)

You can also manually set the `frameRate `value in the console localStorage object. Just make sure to refresh the page to see it in action.

Board Settings are a high priority task for me, so you should be seeing more development in this area coming soon. This is our first mockup (which includes a slider). Feel free to offer up your comments and suggestions.
![board-settings-mockup-01](https://user-images.githubusercontent.com/2898748/139943442-39b364d4-dd65-49b9-8dcb-e081e886f796.png)

You can also poke around my PR/commits to investigate the code. I added a slider initially but it was too clunky and needs more TLC, so in the meantime I added a select so we have some more time to polish the slider: https://github.com/BattlesnakeOfficial/board/pull/40/commits/dc40d4c8beddfcd94078334ed66c49dbd559111f

**Reply by BoldBigflank** [reply]
Did this leak early? I hadn't messed with it but the speed is pretty quick now. I checked my localStorage and have a frameRate key set to 20.
I was checking out 
https://play.battlesnake.com/g/d9f74cc2-a952-4a14-9361-666fa2bb6ebc/
if that matters at all

**Reply by originalwebgurl** [reply]
So it looks like there was a disconnect between the documentation and the older code for `frameRate`. 
The documented default `frameRate ` is 20, however it looks like the actual default that was previously coded for frameRate was 10. 

When I pushed my recent changes, I synced the default with the documented frameRate of 20. So that would cause snakes to move faster.

I'll toss this in front of the team to see if we want to keep the default at 20 or switch back to 10. In the meantime you can change the localStorage value to 10 to see your snakes moving at the speed they used to go.

**Reply by BoldBigflank** [reply]
please make the default 10, 20 is too difficult to follow.

**Reply by originalwebgurl** [reply]
Change made and deployed: https://github.com/BattlesnakeOfficial/board/releases/tag/1.1.2 

You'll need to clear the localStorage object for the board to see the change.

**Comment by originalwebgurl** [comment]
Just thought I'd keep you updated on the board settings progress. My latest PR includes a change that will allow you to properly navigate to the settings view: https://github.com/BattlesnakeOfficial/board/pull/42

![settings-button](https://user-images.githubusercontent.com/2898748/142449026-cde9b8b6-65bd-4e9c-9fea-d3c056a843e7.gif)

**Comment by originalwebgurl** [comment]
Hi @bcambl, this has been fixed and deployed in our latest version: https://github.com/BattlesnakeOfficial/board/releases/tag/1.2.0 

You can read about the [Playback Settings in our latest blog announcement](https://blog.battlesnake.com/playback-settings/).

Thanks for opening this feature request.

**Reply by bcambl** [reply]
Awesome! Thank you for working on this feature. :)
